src/RetrySettings.php

```diff
@@ -235,25 +233,44 @@ class RetrySettings
      */
     private ?Closure $retryFunction;
 
+    /**
+     * When set, this function will be used to evaluate the dealy between
+     * retries. This overrides $retryableCodes, $initialRetryDelayMillis,
+     * and $retryDelayMultiplier.
+     * The callable will have the following signature:
+     * function (int attempts, Exception $e): int (delay in milliseconds)
+     */
+    private ?Closure $retryDelayFunction;
+
+    /**
+     * The deadline in milliseconds.
+     */
+    private ?float $deadlineMillis;
+
+    /*
+     * The number of retries that have already been attempted.
+     * The original API call will have $retryAttempts set to 0.
+     */
+    private int $retryAttempts;
```

Comment:
This object represents the settings for retrying, or _HOW_ the retrying will be performed.

`$retryAttempts` is related to the actual retrying taking place. This should be handled by the _Retrier_, and NOT the RetrySettings.